src/widget/wcuemenupopup.cpp

```diff
@@ -135,6 +186,63 @@ void WCueMenuPopup::slotDeleteCue() {
     hide();
 }
 
+void WCueMenuPopup::slotSavedLoopCue() {
+    VERIFY_OR_DEBUG_ASSERT(m_pCue != nullptr) {
+        return;
+    }
+    VERIFY_OR_DEBUG_ASSERT(m_pTrack != nullptr) {
+        return;
+    }
+    VERIFY_OR_DEBUG_ASSERT(m_pBeatLoopSize.valid()) {
+        return;
+    }
+    if (m_pCue->getType() == mixxx::CueType::Loop) {
+        m_pCue->setType(mixxx::CueType::HotCue);
+        m_pCue->setEndPosition(mixxx::audio::FramePos());
+    } else {
+        double beatloopSize = m_pBeatLoopSize.get();
+        const mixxx::BeatsPointer pBeats = m_pTrack->getBeats();
+        if (beatloopSize <= 0 || !pBeats) {
+            m_pCue->setEndPosition(mixxx::audio::FramePos());
+            return;
+        }
+        m_pCue->setEndPosition(pBeats->findNBeatsFromPosition(m_pCue->getPosition(), beatloopSize));
+        m_pCue->setType(mixxx::CueType::Loop);
+    }
+    slotUpdate();
+}
+
+void WCueMenuPopup::slotAdjustSavedLoopCue() {
+    VERIFY_OR_DEBUG_ASSERT(m_pCue != nullptr) {
+        return;
+    }
+    VERIFY_OR_DEBUG_ASSERT(m_pTrack != nullptr) {
+        return;
+    }
+    VERIFY_OR_DEBUG_ASSERT(m_pBeatLoopSize.valid()) {
+        return;
+    }
+    if (m_pCue->getType() == mixxx::CueType::Loop) {
```

Comment:
I've expected that right-click turns it into a loop cue right away.
I.e. make righ-click use the current position, regardless the current cue type.